bohr/pipeline/apply_heuristics.py

```diff
@@ -9,88 +9,96 @@
 
 import numpy as np
 import pandas as pd
-from snorkel.labeling import PandasLFApplier
+import dask.dataframe as dd
+from dask.diagnostics import ProgressBar
+import csv
+
+from snorkel.labeling import PandasLFApplier, LFAnalysis
+from snorkel.labeling.apply.dask import DaskLFApplier, PandasParallelLFApplier
 
 from snorkel.labeling.model import MajorityLabelVoter
 
-from bohr.heuristics.bugs import bugs, nonbugs
+from bohr.heuristics import all_lfs
+import bohr.heuristics.bug as bug_heuristics
 from bohr.snorkel_utils import BUG, BUGLESS
-from bohr import PROJECT_DIR
-
-def clean_text_columns(df: pd.DataFrame):
-    for column in ["commit_message_stemmed", "issue_contents_stemmed"]:
-        df[column].replace({'\n': ' '}, inplace=True, regex=True)
-        df[column].replace({'\r': ' '}, inplace=True, regex=True)
-        df[column].replace({'-': ' '}, inplace=True, regex=True) # jira regex does not work anymore with this preprocessing step
-        df[column].replace({'/': ' '}, inplace=True, regex=True)
-        df[column].replace({'_': ' '}, inplace=True, regex=True)
-        df[column].replace({'\*': ' '}, inplace=True, regex=True)
-        df[column].replace({r'\\': ' '}, inplace=True, regex=True)
-        df[column].replace({'    ': ' '}, inplace=True, regex=True)
-        df[column].replace({'"': ''}, inplace=True, regex=True)
-        df[column].replace({'\'': ''}, inplace=True, regex=True)
-        punctuation_signs = list("?:!.,;")
-        for punct_sign in punctuation_signs:
-            df[column] = df[column].str.replace(punct_sign, '')
-        df.replace(np.nan, '', inplace=True, regex=True)
-
-
-def majority_acc(L: np.ndarray, df: pd.DataFrame) -> float:
+from bohr import PROJECT_DIR, TRAIN_DIR, TEST_DIR
+
+
+def majority_acc(L: np.ndarray, df: dd.DataFrame) -> float:
     majority_model = MajorityLabelVoter()
-    maj_model_train_acc = majority_model.score(L=L, Y=df.label, tie_break_policy="random")["accuracy"]
+    maj_model_train_acc = majority_model.score(
+        L=L, Y=df.bug, tie_break_policy="random")["accuracy"]
     return maj_model_train_acc
 
 
 def apply_heuristics(args) -> Dict[str, Any]:
     stats: Dict[str, Any] = {}
-    train_filename = PROJECT_DIR / args.dataset_path_train
-    test_filename = PROJECT_DIR / args.dataset_path_test
 
-    df_train = pd.read_csv(train_filename, sep=';', index_col=0, nrows=args.rows_train)
-    df_test = pd.read_csv(test_filename, sep=';', index_col=0, nrows=args.rows_test)
+    df_train = pd.read_csv(TRAIN_DIR / 'b_b.csv', nrows=args.rows_train)
+    df_test = pd.concat([pd.read_csv(p)
+                         for p in TEST_DIR.glob('*.csv')], ignore_index=True, sort=True)
```

Comment:
Do you have any idea why we now extract only 5214 commits instead of 8958 from the dataset of Herzig et al, which results in total 6738 commits instead of 10481 in the training set?

Reply:
Sorry for the late response, there was a deadline today.
Anyway, this is the code: https://github.com/giganticode/bohr-dataset/blob/master/build_test.py#L64
So, seems indeed like I filtered out anything that cannot be related to a GitHub commit. The unfiltered set, however,
has only 7406 rows, so that's strange...

I remember there being several versions of the Herzig set, maybe there is a larger one?